compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -915,18 +913,24 @@ pub trait PrintState<'a>: std::ops::Deref<Target = pp::Printer> + std::ops::Dere
             token::Literal(lit) => literal_to_string(lit).into(),
 
             /* Name components */
-            token::Ident(s, is_raw) => {
-                IdentPrinter::new(s, is_raw.into(), convert_dollar_crate).to_string().into()
+            token::Ident(name, is_raw) => {
+                IdentPrinter::new(name, is_raw.into(), convert_dollar_crate).to_string().into()
             }
-            token::Lifetime(s) => s.to_string().into(),
+            token::InterpolatedIdent(ident, is_raw) => {
+                IdentPrinter::new(ident.name, is_raw.into(), convert_dollar_crate)
```

Comment:
```suggestion
                IdentPrinter::for_ast_ident(ident, is_raw.into()).to_string()
```
Pure refactoring would look like this ^.
I think the new version does something slightly different. (Different span is used in the last argument?)

Reply:
True. Both versions have the same behaviour w.r.t. the test suite, so I'm happy to use `for_ast_ident`.